src/RestServer/Controllers/v1/ContractsController.cs

```diff
@@ -0,0 +1,217 @@
+        [HttpGet(Name = "GetContracts")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(ContractState[]))]
+        public IActionResult Get(
+            [FromQuery(Name = "page")]
+            int skip = 1,
+            [FromQuery(Name = "size")]
+            int take = 50)
+        {
+            if (skip < 1 || take < 1 || take > RestServerSettings.Current.MaxPageSize)
+                throw new InvalidParameterRangeException();
+            var contracts = NativeContract.ContractManagement.ListContracts(_neosystem.StoreView);
+            if (contracts.Any() == false)
+                return NoContent();
+            var contractRequestList = contracts.OrderBy(o => o.Manifest.Name).Skip((skip - 1) * take).Take(take);
+            if (contractRequestList.Any() == false)
+                return NoContent();
+            return Ok(contractRequestList);
+        }
+
+        /// <summary>
+        /// Gets count of total smart contracts on blockchain.
+        /// </summary>
+        /// <returns>Count Object</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("count", Name = "GetContractCount")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(CountModel))]
+        public IActionResult GetCount()
+        {
+            var contracts = NativeContract.ContractManagement.ListContracts(_neosystem.StoreView);
+            return Ok(new CountModel() { Count = contracts.Count() });
+        }
+
+        /// <summary>
+        /// Get a smart contract's storage.
+        /// </summary>
+        /// <param name="scripthash" example="0xed7cc6f5f2dd842d384f254bc0c2d58fb69a4761">ScriptHash</param>
+        /// <returns>An array of the Key (Base64) Value (Base64) Pairs objects.</returns>
+        /// <response code="200">Successful</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpGet("{hash:required}/storage", Name = "GetContractStorage")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(KeyValuePair<ReadOnlyMemory<byte>, ReadOnlyMemory<byte>>[]))]
+        public IActionResult GetContractStorage(
+            [FromRoute(Name = "hash")]
+            UInt160 scripthash)
+        {
+            if (NativeContract.IsNative(scripthash))
+                return NoContent();
+            var contract = NativeContract.ContractManagement.GetContract(_neosystem.StoreView, scripthash);
+            if (contract == null)
+                throw new ContractNotFoundException(scripthash);
+            var contractStorage = contract.GetStorage(_neosystem.StoreView);
+            return Ok(contractStorage.Select(s => new KeyValuePair<ReadOnlyMemory<byte>, ReadOnlyMemory<byte>>(s.key.Key, s.value.Value)));
```

Comment:
The server will die on attempt to return the whole storage of some large contract. We need restrictions for this method, paging will suite.